### PR TITLE
Added Rotation Of Numbers

### LINKED ISSUE #1421: Program for generating all rotations of a number
In c, c++, java .python and go

### PR DESCRIPTION
**Pull request by somya-kapoor**
Fixes #1421 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added Rotation Of Numbers

#### Languages Used:
- C
- C++
- Java
- Python

#### Files Added:
- Rotation_Of_Numbers.c 
- Rotation_Of_Numbers.cpp
- Rotation_Of_Numbers.java
- Rotation_Of_Numbers.py

### CODE REVIEW

File: Rotation_Of_Numbers/Rotation_Of_Numbers.c [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#include<stdio.h>
+
+// Function to rotations of a number 
+void rotation (int num)
```
Comment (marwahmanbir) [inline Rotation_Of_Numbers/Rotation_Of_Numbers.c]: Remove space after ``rotation``.

File: Rotation_Of_Numbers/Rotation_Of_Numbers.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    
+    int power = pow(10, digits - 1); 
+    
+    // Rotations of the number
+    for (int i = 0; i < digits - 1; i++) 
+    {
+        int lastDigit = num % 10; 
+        int x = num / 10;
+        num = lastDigit * power + x;
+
```
Comment (marwahmanbir) [inline Rotation_Of_Numbers/Rotation_Of_Numbers.c]: Remove extra blank line.

File: Rotation_Of_Numbers/Rotation_Of_Numbers.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#include<bits/stdc++.h> 
+using namespace std; 
+  
+// Function to rotations of a number 
+void rotation (int num)
```
Comment (marwahmanbir) [inline Rotation_Of_Numbers/Rotation_Of_Numbers.cpp]: Remove space after ``rotation``.

File: Rotation_Of_Numbers/Rotation_Of_Numbers.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    
+    int power = pow(10, digits - 1); 
+    
+    // Rotations of the number
+    for (int i = 0; i < digits - 1; i++) 
+    {
+        int lastDigit = num % 10; 
+        int x = num / 10;
+        num = lastDigit * power + x;
+
```
Comment (marwahmanbir) [inline Rotation_Of_Numbers/Rotation_Of_Numbers.cpp]: Remove

File: Rotation_Of_Numbers/Rotation_Of_Numbers.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import java.util.*;
+
+class Rotation_Of_Number
+{ 
+    // Function to rotations of a number 
+    static void rotation (int num)
```
Comment (marwahmanbir) [inline Rotation_Of_Numbers/Rotation_Of_Numbers.java]: Fix here also.

File: Rotation_Of_Numbers/Rotation_Of_Numbers.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+        int power = (int) Math.pow(10, digits - 1); 
+
+        // Rotations of the number
+        for (int i = 0; i < digits - 1; i++) 
+        {
+            int lastDigit = num % 10; 
+            int x = num / 10;
+            num = lastDigit * power + x;
+
```
Comment (marwahmanbir) [inline Rotation_Of_Numbers/Rotation_Of_Numbers.java]: Remove.

**Comment by somya-kapoor** [comment]
@marwahmanbir Done, please review

**Pull request merged**